Always update formController for viewmodels when singleton is updated

lognaturel opened the pull request:
Closes #4327 

Some users are getting a crash that suggests their singleton `formController` is set but the `ViewModel`s don't know about it.

I searched for `Collect.getInstance().setFormController` and made sure that when it's called, `formControllerAvailable` is always called with the same value.

I also added commits improving logging that I saw while troubleshooting this.

#### What has been done to verify that this works as intended?
I can't reproduce the issue so all I have done is visual inspection.

#### Why is this the best possible solution? Were any other approaches considered?
It doesn't make sense that those two updates would be different so I think this is a change we have to make. I also updated some logging. The other approach I considered was passing in `formController` to `moveForward` and `moveBackward` and logging if the view model one is null but the one passed in is not. I think that would be fine and perhaps we can do that too but I think this change actually addresses the root cause.

I initially wanted to also null out the `ViewModel`s' `formController`s when `           Collect.getInstance().setFormController(null)` is called. However, that's not necessary for this fix and would probably require introducing new methods for those view models. It feels weird that there might be a stale form controller set but we can come back to it.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
I believe this to be very low risk. We are calling `formControllerAvailable` in more cases which seems it would always be positive.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

lognaturel committed:
Always update formController for viewmodels when singleton is updated

lognaturel commented on collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:
```diff
@@ -2297,7 +2301,7 @@ public void denied() {
                     } catch (Exception e) {
                         // if somehow we end up with a bad language, set it to the default
-                        Timber.e("Ended up with a bad language. %s", newLanguage);
+                        Timber.e(e, "Ended up with a bad language. %s", newLanguage);
```
I see this in a lot of our logs and I can't tell if it's actually meaningful. Hopefully seeing the exception will give us some insights. I'm hoping it turns out to be expected and we can just remove the logging.

grzesiek2010 replied:
It takes place every time I open the demo form. I think it's caused by language code, we have just label but it should be label::English (en) https://xlsform.org/en/#multiple-language-support

lognaturel replied:
Yeah, I saw the stacktrace locally and realized it's really nothing so I should take this change out. 

It doesn't have to do with language codes because those are ignored in JR/Collect for now. I think it's just that because no language was explicitly chosen from the menu the database column for language has a null value.

lognaturel replied:
To be explicit, I think it's important to clarify/clean up this logging because it can waste our time when troubleshooting logs on Crashlytics.

lognaturel committed:
Add more logging for attachControlsToInstanceData crash

lognaturel committed:
Improve ContentUriProvider logging

lognaturel committed:
Don't remotely log bad language

This is guaranteed to happen any time a language hasn't been explicitly set by the user.

grzesiek2010 commented on collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:
```diff
@@ -2282,6 +2283,8 @@ public void denied() {
                 t.destroy();
 
                 Collect.getInstance().setFormController(formController);
+                formControllerAvailable(formController);
```
This change makes sense. I wonder if we should also maybe dismiss the loading dialog later once we set everything because now we do that at the beginning of this method what unblocks the UI and if the rest of the job we do in this method takes some time it's dangerous.
